hw1/bashcommands.py

```diff
@@ -0,0 +1,79 @@
+from collections import deque
+
+
+def for_each_argument(f):
+    def f_for_many(*many_args):
+        for arg in many_args:
+            f(arg)
+
+    return f_for_many
+
+
+def wc(*filepaths):
+    total_lines = 0
+    total_words = 0
+    total_bytes = 0
+    for filepath in filepaths:
+        cur_lines = 0
+        cur_words = 0
+        cur_bytes = 0
+        try:
+            with open(filepath) as f:
+                for line in f:
+                    cur_lines += 1
+                    cur_words += len(line.split())
+                    cur_bytes += len(bytes(line, encoding="utf8"))
+            total_lines += cur_lines
+            total_words += cur_words
+            total_bytes += cur_bytes
+            print(f"lines:{cur_lines} words:{cur_words} bytes:{cur_bytes} for {filepath}")
+        except FileNotFoundError:
+            print(filepath, "no such file")
+    print(f"lines:{total_lines} words:{total_words} bytes:{total_bytes} in total\n")
+
+
+@for_each_argument
+def nl(filepath):
+    print(f"==> {filepath} <==")
```

Comment:
Nice touch, but `nl` actually doesn't do this. And it actually keeps the numbering. So when one file ends, the next one just starts at the next number. So the indexes must be continuous for all files.

Reply:
(for `head` and `tail`, the use of "==>" is correct, yes)

Reply:
Also, `nl` skips empty lines, but prints them. Try running it on empty lines to catch it.